arroyo/processing/strategies/filter.py

```diff
@@ -28,8 +39,30 @@ def poll(self) -> None:
     def submit(self, message: Message[TPayload]) -> None:
         assert not self.__closed
 
+        now = time.time()
+
         if self.__test_function(message):
+            if self.__commit_policy_state is not None:
+                self.__flush_uncommitted_offsets(now)
             self.__next_step.submit(message)
+        elif self.__commit_policy_state is not None:
+            self.__uncommitted_offsets.update(message.committable)
+
+            if self.__commit_policy_state.should_commit(now, message.committable):
```

Comment:
Could we move the `should_commit` check into the `__flush_uncommitted_offsets` method. That way, even in the other code branch (test function passes and message is valid) we still keep the same throttling behavior